src/NexusMods.App.UI/Helpers/DoubleClickHelper.cs

```diff
@@ -0,0 +1,117 @@
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Subjects;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using JetBrains.Annotations;
+
+namespace NexusMods.App.UI.Helpers;
+
+[PublicAPI]
+public static class DoubleClickHelper
+{
+    // NOTE(erri120): Windows uses 500ms (https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setdoubleclicktime?redirectedfrom=MSDN#parameters)
+    // This can be fine-tuned later.
+    public static readonly TimeSpan DefaultTimeout = TimeSpan.FromMilliseconds(500);
+
+    /// <summary>
+    /// Registers a double click handler and returns an observable that triggers whenever
+    /// a double click occurs.
+    /// </summary>
+    public static IObservable<Unit> AddDoubleClickHandler(this Control control)
+    {
+        return AddDoubleClickHandler(control, InputElement.PointerPressedEvent);
+    }
+
+    /// <summary>
+    /// Registers a double click handler and returns an observable that triggers whenever
+    /// a double click occurs.
+    /// </summary>
+    public static IObservable<Unit> AddDoubleClickHandler<TRoutedEventArgs>(this Control control, RoutedEvent<TRoutedEventArgs> routedEvent)
+        where TRoutedEventArgs : RoutedEventArgs
+    {
+        return new DoubleClickSubject<TRoutedEventArgs>(control, routedEvent);
+    }
+
+    private class DoubleClickSubject<TRoutedEventArgs> : SubjectBase<Unit>
+        where TRoutedEventArgs : RoutedEventArgs
+    {
+        private bool _isDisposed;
+
+        private readonly List<IObserver<Unit>> _observers = new(capacity: 1);
+        private readonly IDisposable _handlerDisposable;
+
+        public DoubleClickSubject(Control control, RoutedEvent<TRoutedEventArgs> routedEvent)
+        {
+            _handlerDisposable = control.AddDisposableHandler(
+                routedEvent,
+                EventHandler,
+                routes: RoutingStrategies.Bubble,
+                handledEventsToo: true
+            );
+        }
+
+        private DateTime _lastDateTime = DateTime.UnixEpoch;
+        private void EventHandler(object? sender, TRoutedEventArgs args)
+        {
+            var currentDateTime = DateTime.Now;
+            var isDoubleClick = currentDateTime - _lastDateTime <= DefaultTimeout;
+            _lastDateTime = currentDateTime;
+
+            if (!isDoubleClick) return;
+            OnNext(Unit.Default);
+            args.Handled = true;
+        }
```

Comment:
I found this: https://github.com/AvaloniaUI/Avalonia/discussions/11922
Where they suggest using the `ClickCount` property of the event to handle double clicks.

There also appears to be a `DoubleTapped` event, though I don't know if all controls support that:
https://docs.avaloniaui.net/docs/concepts/input/pointer#more-information

Have you already considered those approaches?

Reply:
`ClickCount` is handled not on the event, but on the input device itself:

https://github.com/AvaloniaUI/Avalonia/blob/4fdf4fa36cfc9b1e353a4ca79e5c8e8137bd9ea2/src/Avalonia.Base/Input/MouseDevice.cs#L133-L148

`PointerPressedEventArgs` are also the only event args that have the `ClickCount` property. My approach works for any event, as it just waits for the same event to occur twice within a given interval.

Reply:
I see, this approach is more generalized 👍